src/lib.rs

```diff
@@ -268,21 +271,95 @@ impl<'object, Endian> DebugInfo<'object, Endian>
                 continue;
             }
 
+            let mut rowi = 0;
+            let mut current = None;
+
             // Okay, this is the right unit. Check our DebugLine rows.
-            let mut rows = unit.line_rows(&self.debug_line)
-                .chain_err(|| "cannot get line rows for unit")?;
+            let mut rows = {
+                // First, check our binary search list
+                if let Ok(skiplist) = unit.skiplist.read() {
+                    match skiplist.binary_search_by_key(&addr, |&(raddr, _, _)| raddr) {
+                        Ok(i) => {
+                            // we have a state machine for this address!
+                            current = Some(skiplist[i].2);
+                            rowi = (i + 1) * self.cache_every;
```

Comment:
The skiplist holds the state machine from after reading a row that has an index that is a multiple of `cache_every`, which means the next row that state machine gives will be one more than that, so this should be `rowi = (i + 1) * self.cache_every + 1;`.